### PR TITLE
(Fixes #239) Capability to generate graph of workflow

### LINKED ISSUE #239: Capability to generate graph of workflow

### PR DESCRIPTION
**Pull request by sethyates**

### CODE REVIEW

File: core/src/main/scala/com/krux/hyperion/workflow/WorkflowGraphRenderer.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import com.krux.hyperion.DataPipelineDef
+
+import scala.util.Try
+
+case class WorkflowGraphRenderer(
+  pipeline: DataPipelineDef,
+  removeLastNameSegment: Boolean,
+  includeResources: Boolean
+) {
+  private lazy val styles = ConfigFactory.load("reference").getConfig("hyperion.graphviz.styles")
```
Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/workflow/WorkflowGraphRenderer.scala]: `ConfigFactory.load` should be good enough, right? it doesn't have to be `"reference"`

Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/workflow/WorkflowGraphRenderer.scala]: `load` loads the resource.

Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/workflow/WorkflowGraphRenderer.scala]: Actually, I think should get this from `HyperionContext`, so we have a centralized place for getting all these configurations.

Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/workflow/WorkflowGraphRenderer.scala]: And how does one do that?

Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/workflow/WorkflowGraphRenderer.scala]: You already `pipeline: DataPipelineDef` in `WorkflowGraphRenderer`
```
val style = pipeline.hc.graphStyle
```

Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/workflow/WorkflowGraphRenderer.scala]: `pipeline.hc`?

Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/workflow/WorkflowGraphRenderer.scala]: OK.

File: core/src/main/scala/com/krux/hyperion/workflow/WorkflowGraphRenderer.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  private def renderEdge(from: String, to: String) = {
+    val attrs = getAttributes(s"${idToTypeMap(from)}To${idToTypeMap(to)}", styles)
+      .orElse(getAttributes(s"${idToTypeMap(from)}ToAny", styles))
+      .orElse(getAttributes(s"AnyTo${idToTypeMap(to)}", styles))
+    s"  ${quoted(from)} -> ${quoted(to)} ${attrs.getOrElse("")}"
+  }
+
+  def render(): String = {
+    val parts = Seq(s"strict digraph ${quoted(pipeline.pipelineName)} {") ++
+    pipelineObjects.flatMap { obj =>
```
Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/workflow/WorkflowGraphRenderer.scala]: need more indentation

**Comment by realstraw** [comment]
:+1: :sparkles: :sparkles: :sparkles:

**Pull request closed without merging**